### PR TITLE
Remove legacy steps

### PR DESCRIPTION
**Pull request by mwhipple**
These have been disabled in recent versions, and are being
removed to ease migration to ParameterTypes.

### CODE REVIEW

**Comment by mwhipple** [comment]
There is some code in util.rb that is likely also dead now.

**Comment by mwhipple** [comment]
There is some code in util.rb that is likely also dead now.

File: features/request_construction/basic_auth.feature [hunk 1]
```diff
@@ -5,7 +5,7 @@ Feature: Adding Basic Auth
       """
       {"Authorization":"Basic dXNlcjpwYXNz"}
       """
-    And expected GET sent to `/query` with
```
Comment (jackterranova) [inline features/request_construction/basic_auth.feature]: Was this previously mapped in rest_steps?

Comment (mwhipple) [reply features/request_construction/basic_auth.feature]: This was previously sloppily matched due to the missing anchor here: https://github.com/brightcove/brine/pull/194/files#diff-688e9e1d81dcfc63a49490439ed1b02fL124

**Review by jackterranova — APPROVED** [review]

**Pull request merged**